rpc/client.go

```diff
@@ -1,33 +1,72 @@
 package rpc
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"runtime"
+	"strconv"
 	"sync"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
 	"github.com/evergreen-ci/juniper/gopb"
 	"github.com/evergreen-ci/poplar"
 	"github.com/evergreen-ci/poplar/rpc/internal"
+	"github.com/google/uuid"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
 	"github.com/mongodb/grip/recovery"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 )
 
+// UploadReportOptions contains all the required options for uploading a report
+// to both Cedar and the S3 data-pipes service.
```

Comment:
I don't think we should specify S3 here and instead just say `Data Pipes service`.